Profile Update Shelly 1PM - Adding sub-profiles

dagobert opened the pull request:
I added sub-profiles and re-done all the measurements. The previously measured standby powers were not mrecise because the measuring device of the contributor needs more dummy power usage to be precise. For that measuring device I revommend at least 7W of dummy power. Below I noticed larger variations in measurements.

dagobert commented:
Just for my interest, what happens to anyone already  using the profile after updating zmto this new version with aub-profiles?

bramstroker commented:
I just thought about that as well, that would be an issue.

I will come back on this later, to busy atm.
But will revert this PR for now

Mariusthvdb commented:
Thanks for adding these, it was the final profile I missed, and I still used this in y yaml:
```
- entity_id: switch.slaapkamer_stopcontact
  standby_power: 0.5
  fixed:
    power: 0.8
```

which were my measurements at the time I installed it first.

please let me ask here even while closed:

did you check the various BT modes on the device?

Tbh, the difference between the 2 specified modes using BT or not is so small, this seems to overcomplicate things unnecessarily, and based on metering variances. Really think you should not offer those 2 as separate profiles because of that.

The default profile consuming almost double is also a wild guess? or did you actually measure that.

On these devices the below choice would probably be most interesting, although I fear even that will be difficult to measure:

<img width="595" alt="Scherm­afbeelding 2025-03-09 om 16 10 58" src="https://github.com/user-attachments/assets/f363e816-bd22-47e8-ab39-e2a11723cf3f" />

The other main difference I can imagine is whether the device is used as Wifi AP or not, but I dont think you measured that either?

Checking the device settings on the device does not mention WLAN at all, so why did you chose to use that for profile naming, it is rather confusing tbh.

dagobert commented:
> ```
> - entity_id: switch.slaapkamer_stopcontact
>   standby_power: 0.5
>   fixed:
>     power: 0.8
> ```

Hi,
Great we can help!
You can use the profile without a problem. You yaml code should have listed 0.8 for standby_power _on, just as a side note.

> did you check the various BT modes on the device?
> 
> Tbh, the difference between the 2 specified modes using BT or not is so small, this seems to overcomplicate things unnecessarily, and based on metering variances. Really think you should not offer those 2 as separate profiles because of that.
> The other main difference I can imagine is whether the device is used as Wifi AP or not, but I dont think you measured that either?

I have done the extensive power measurements for some shelly devices and added them as sub-profiles. You can see an example with the the [Shelly 1PM mini Gen3](https://github.com/bramstroker/homeassistant-powercalc/tree/master/profile_library/shelly/Shelly 1PM Mini Gen3). I obseved that you only need to do six measurements in three device settings.

1. Wifi turned and ECO turned on, all else turned off.
2. Wifi, Bluetooth and ECO turned on.
3. ECO turned off with any other settings.

These three settings you need to measure the device with the switch turned off and than turned on (if it has a switch / relay).

If you do these six measurements, I am happy to create or enhance the profile for you.

Mariusthvdb commented:
> You yaml code should have listed 0.8 for standby_power _on, just as a side note.

why do you think so? it is what I measured at the time.

I suppose I feel differently about the scenarios 1 and 2 in the measurements. They are identical. No use to distinguish between 0,05 watt if you ask me, as that can not be measured reliably.

 can see the difference between eco on and eco off, so my suggestion would be to only list those 2 scenarios, unless the different BT modes in my screenshot would change stuff significantly.

But, those are also not in your measurements are they? I can no longer measure them, as the device is already inside the wall ;-)

If you would be able to check that, that could be useful to finetune the profile for this device.

dagobert commented:
> > You yaml code should have listed 0.8 for standby_power _on, just as a side note.
> 
> why do you think so? it is what I measured at the time.

*Fixed* power consumption is used to define the usage of the device connected to, in this case, Shelly 1PM. The Shelly device is measuring power of the devices connected to it. Hence, you do not define fixed power. The fixed power you defined is the *standby_power_on* usage of the Shelly device itself. 
I did not doubtyour measurement. Sorry, if this was unclear.

> I suppose I feel differently about the scenarios 1 and 2 in the measurements. They are identical. No use to distinguish between 0,05 watt if you ask me, as that can not be measured reliably.

It is possible that your measurements did not find any notably difference (>0.1W) between settings 1. and 2. Depending on your measuring setup it is not easy or possible to measure such small amount of power usage. Using the "Shelly PM Mini Gen3" as one recommendation of the powercalc maintainer is delivering reasonable results **if** you use a good non fluctuation dummy power of at least 7W.

> The difference between eco on and eco off seems obvious, so my suggestion would be to only list those 2 scenarios, unless the different BT modes in my screenshot would change stuff significantly.

As mentioned, I have measured multiple different Shelly devices and always found a sigificant difference between these three settings - actually two and ECO off. It is certainly possible that it is different with the "Shelly 1PM" device. 

> But, those are also not in your measurements are they? I can no longer measure them, as the device is already inside the wall ;-)

No, I did extensive testing measurements with all different kind of settings, namely wifi, wifi ap, bt, bt extender, eco and so on in different combinations. The three different profiles were the only ones that resulted in different measurements.

> If you would be able to check that, that could be useful to finetune the profile for this device.

As a side note, this PR is closed, because the changes of the profile were wrongly attributed by me. I created the PR but noticed later that I had measured the Shelly Plus 1PM. There is not updated measurement for this device.
If you delete your yaml code you should still benefit from the current profile of this device ans further enhancements if they are submitted.

dagobert commented:
> cant find the documentation on the standby_power_on setting in https://docs.powercalc.nl/strategies/fixed/, so for now I changed it to
> 
> ```
>         - entity_id: switch.slaapkamer_stopcontact
>           fixed:
>             states_power:
>               'on': 0.8
>               'off': 0.5
> ```

The *fixed* setting is only for power usage of the lamp or electrical device connected to the Shelly. The power draw of the shelly is accounted for with the *standby_...* settings found [here](https://docs.powercalc.nl/configuration/standby-power/).

> 
> this is with BT on (active), Eco on, and No Wifi acces point)
> 
> as for the measurements, Ill leave it at that, because I seem to not be able to express what I mean sufficiently ;-) to me the difference between:
> 
> ```
> {
>   "name": "WLAN + Bluetooth + ECO",
>   "standby_power": 0.32,
>   "standby_power_on": 0.54
> }
> ```
> 
> and
> 
> ```
> {
>   "name": "WLAN + ECO",
>   "standby_power": 0.31,
>   "standby_power_on": 0.56
> }
> ```
> 
> is non existent.

Yes you are right. This is not really a difference. The measurements I posted were done in a batch and with this device I had a mixup it seem, as mentioned. Since I noticed that I have not checkt the changes for other errors.  If you have a look at [this](https://github.com/bramstroker/homeassistant-powercalc/tree/master/profile_library/shelly/Shelly%201PM%20Mini%20Gen3) you can see how much the differences probably would be. Just forget this PR. 

Thanks for hinting me to this issue with the already submitted power profile. I will have to do the measurements of the *Shelly Plus 1PM* again since my kids already shredded my notes, as they love to do ;)

> the word choice for standby is wrong imo, the switch is on/off, no matter the other possible configuration options..... maybe that is semantics, so nevermind.

Yes, choosing the "right" words is often a matter of optinion and I can agree with you. Just understand that this software has grown and when you start with implementing eg. with smart light in mind, a light turned off still has a power draw. How to name this? Yes, the light is still on from a technical perspective and looking at the controm electronics. From the user view just looking at it it is off. Same with the TV - when it is not showing a picture, it is off and toway of cause it is in standby mode. The power draw is than specified by the different calcualtion options linear, fixed or lut.

Smart switches therefore are always in standby for powercalc since they are not electrical devices for the user visible to be on or off.
To account for the Shelly you use standby_power for its off state and standby_power_on for its on state, hence to account for the lamp connected you use eg. fixed.

> the above manual config does not behave properly for the 'off' state, as the measurement then drops to 0. So, I took it back to the original
> 
> ```
>         - entity_id: switch.slaapkamer_stopcontact
>           standby_power: 0.5
>           fixed:
>             power: 0.8
> ```
> 
> and that does what is supposed to do:
> including the top (actueel, power) and bottom (kWh) for their measured power delivery to the plugged in vacuum ;-)

Yes of cause it does. This Shelly device does have its own power measurement. *Fixed* is active when the Shelly switch is "On" and *standby_power* is not active any more. In your logic the end device you connected to the Shelly uses 0.8W when it is "running". But imagine you would have a Shelly without measurement capability. Than the fixed value would account for its power draw, meaning it would be eg. 20W. But not you are missing the higher power draw if the Shellly device itself for holding the relay in state "On" - your measured 0.8W.

I recommend you to remove the YAML code and use the profile from the library which is [this](https://github.com/bramstroker/homeassistant-powercalc/blob/master/profile_library/shelly/Shelly%201PM/model.json) one. It has basically the same values as your measurements and will work the same as yours.

I hope with this extensive post I was able to clarify some missunderstandings. Again, I really understand your point and am thankful for noticing the values of the other model, which I will recheck.
